group/assignment6/TestRearrangeCars.cpp

```diff
@@ -0,0 +1,30 @@
+#include "TestRearrangeCars.h"
+
+void TestRearrangeCars::runAllTests() {
+	testValidRearrangingOfCars();
+}
+
+template<typename T>
+void printSequence(vector<T> sequence){
+	for (T &item : sequence)
+		cout << item;
+}
+
+void TestRearrangeCars::testValidRearrangingOfCars() {
+	vector<int> startStates = { 0, 1, -1, 2 };

```

Comment:
I started to draft some test cases, and for in case of empty startStates and/or endStates vectors the code fails, I suppose there should be some check for that. Same for the case when startStates and endStates have different lengths. Same when there are several empty slots in startStates and/or endStates, same when there are duplicates in startStates and/or endStates. Of course we can fix it later, when all the tests will be ready. Maybe we should add some additional method to valudate input vectors?

Reply:
I agree. We can add methods for that